extensions/cache/src/main/java/org/radargun/stages/cache/generators/RandomStreamGenerator.java

```diff
@@ -0,0 +1,141 @@
+package org.radargun.stages.cache.generators;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Random;
+
+import org.radargun.config.DefinitionElement;
+
+/**
+ * Generator for {@link RandomStream} objects
+ * 
+ * @author zhostasa
+ */
+@DefinitionElement(name = "randomStream", doc = "Generates stream of random data, specifically for stream operation testing")
+public class RandomStreamGenerator implements ValueGenerator {
+
+   public Object generateValue(Object key, int size, Random random) {
+      return new RandomStream(key.hashCode(), size);
```

Comment:
Why do you map it to the key? When doing repeated writes, we usually want to change the value.